platform/viewer/src/routes/Mode/Mode.tsx

```diff
@@ -38,20 +38,41 @@ async function defaultRouteInit({
 
   unsubscriptions.push(instanceAddedUnsubscribe);
 
-  const { unsubscribe: seriesAddedUnsubscribe } = DicomMetadataStore.subscribe(
-    DicomMetadataStore.EVENTS.SERIES_ADDED,
-    ({ StudyInstanceUID, madeInClient }) => {
-      const studyMetadata = DicomMetadataStore.getStudy(StudyInstanceUID);
-      if (!madeInClient) {
-        HangingProtocolService.run(studyMetadata);
-      }
+  const { unsubscribe: seriesAddedUnsubscribe } = DisplaySetService.subscribe(
+    DisplaySetService.EVENTS.DISPLAY_SETS_CHANGED,
+    displaySets => {
+      if (!displaySets || !displaySets.length) return;
+      // The assumption is that the display set at position 0 is the first
+      // study being displayed, and is thus the "active" study.
+      const studyMap = {};
+      const studies = displaySets.
+        reduce((prev, curr) => {
+          const { StudyInstanceUID } = curr;
+          if (!studyMap[StudyInstanceUID]) {
+            const study = DicomMetadataStore.getStudy(StudyInstanceUID);
+            studyMap[StudyInstanceUID] = study;
+            prev.push(study);
+          }
+          return prev;
+        }, []);
+      const studyMetaData = studies[0];
+      HangingProtocolService.run({ studies, studyMetaData, displaySets });
     }
   );
   unsubscriptions.push(seriesAddedUnsubscribe);
 
+  // No point trying to fire partial retrieves, as it causes havoc
+  // when some series are not displayable, and causes the wrong series
+  // to be fetched.
+  DisplaySetService.holdChangeEvents();
+  const allRetrieves = [];
```

Comment:
Can you make it a `map` instead?

```
const allRetrieves = studyInstanceUIDs.forEach((StudyInstanceUID) => {
  return dataSource.retrieve.series.metadata({ StudyInstanceUID })
})
```

Reply:
Why?  I don't mind doing it, but would like to know why you think that is a good idea.  Objects in Javascript are in fact maps, so I don't see why it is necessary or desirable.

Reply:
with forEach and push you are basically doing what is implemented via map which is transformation of an array as we are doing here